app.go

```diff
@@ -900,16 +900,18 @@ func (app *App) ShutdownWithTimeout(timeout time.Duration) error {
 //
 // ShutdownWithContext does not close keepalive connections so its recommended to set ReadTimeout to something else than 0.
 func (app *App) ShutdownWithContext(ctx context.Context) error {
-	if app.hooks != nil {
```

Comment:
Here we can define executeOnPostShutdownHooks hook with defer as well as PreShutdownHook

Reply:
> Here we can define executeOnPostShutdownHooks hook with defer as well as PreShutdownHook

Is that what you mean?I think `PreShutdownHook `and `PostShutdownHook `set outside the function and then executed at ShutdownWithContext

```go
func (app *App) ShutdownWithContext(ctx context.Context) error {
	app.mutex.Lock()
	defer app.mutex.Unlock()

	if app.server == nil {
		return ErrNotRunning
	}

	// Execute the Shutdown hook
	if app.hooks != nil {
		app.hooks.executeOnPreShutdownHooks()
	}
	defer app.hooks.executeOnPostShutdownHooks(nil)

	return app.server.ShutdownWithContext(ctx)
}
```